pkg/jobs/canary/canary_jobs.go

```diff
@@ -66,54 +65,75 @@ var minimumTimeBetweenCanaryRuns = 10 * time.Second
-		logger.Debugf("config (id=%s) is already running. skipping this run ...", job.Canary.GetPersistedID())
+		logger.Debugf("config (id=%s) is already running. skipping this run ...", canaryID)
 		return
 	}
 	defer lock.Unlock()
 
 	lastRunDelta := minimumTimeBetweenCanaryRuns
 	// Get last runtime from sync map
 	var lastRuntime time.Time
-	if lastRuntimeVal, exists := canaryLastRuntimes.Load(job.Canary.GetPersistedID()); exists {
+	if lastRuntimeVal, exists := canaryLastRuntimes.Load(canaryID); exists {
 		lastRuntime = lastRuntimeVal.(time.Time)
 		lastRunDelta = time.Since(lastRuntime)
 	}
 
 	// Skip run if job ran too recently
 	if lastRunDelta < minimumTimeBetweenCanaryRuns {
-		logger.Infof("Skipping Canary[%s]:%s since it last ran %.2f seconds ago", job.Canary.GetPersistedID(), job.GetNamespacedName(), lastRunDelta.Seconds())
+		logger.Infof("Skipping Canary[%s]:%s since it last ran %.2f seconds ago", canaryID, job.GetNamespacedName(), lastRunDelta.Seconds())
 		return
 	}
 
 	// Get transformed checks before and after, and then delete the olds ones that are not in new set
-	existingTransformedChecks, _ := db.GetTransformedCheckIDs(job.Canary.GetPersistedID())
-	var newChecksCreated []string
+	existingTransformedChecks, _ := db.GetTransformedCheckIDs(canaryID)
+	var transformedChecksCreated []string
+	// Transformed checks have a delete strategy
+	// On deletion they can either be marked healthy, unhealthy or left as is
+	checkIDDeleteStrategyMap := make(map[string]string)
 	results := checks.RunChecks(job.NewContext())
 	for _, result := range results {
 		if job.LogPass && result.Pass || job.LogFail && !result.Pass {
 			logger.Infof(result.String())
 		}
-		checkIDsAdded := cache.PostgresCache.Add(pkg.FromV1(result.Canary, result.Check), pkg.FromResult(*result))
-		newChecksCreated = append(newChecksCreated, checkIDsAdded...)
+		transformedChecksAdded := cache.PostgresCache.Add(pkg.FromV1(result.Canary, result.Check), pkg.FromResult(*result))
+		transformedChecksCreated = append(transformedChecksCreated, transformedChecksAdded...)
+		for _, checkID := range transformedChecksAdded {
+			checkIDDeleteStrategyMap[checkID] = result.Check.GetTransformDeleteStrategy()
+		}
 	}
 	job.updateStatusAndEvent(results)
 
-	// Checks which are not present now should be marked as healthy
-	checksToMarkHealthy := utils.SetDifference(existingTransformedChecks, newChecksCreated)
-	if err := db.UpdateChecksStatus(checksToMarkHealthy, models.CheckStatusHealthy); err != nil {
-		logger.Errorf("error deleting transformed checks for canary %s: %v", job.Canary.GetPersistedID(), err)
+	checkDeleteStrategyGroup := make(map[string][]string)
+	checksToRemove := utils.SetDifference(existingTransformedChecks, transformedChecksCreated)
+	if len(checksToRemove) > 0 && len(transformedChecksCreated) > 0 {
+		for _, checkID := range checksToRemove {
+			strategy := checkIDDeleteStrategyMap[checkID]
+			var status string
+			if strategy == "MarkHealthy" {
```

Comment:
Can we use Constants ?
I think ignore should be the default strategy - i.e. do not insert or update anything, just mark as deleted.

Reply:
👍🏽